fecgov/fec-cms
# Feature/2744 aggregate totals module

## Summary
Updating the aggregate totals module to separate out its CSS and JavaScript and updated Gulp and Webpack to handle those changes. Added a first version of the new monospace font.

- Resolves #2744 
Continuation from PR #2768 

- Moved the aggregate-totals js from components to modules because the name is more accurate.
- Added commands for it to add its own css to its document's head to account for unknown parent urls.
- Removed agg-totals scss from data-landing.
- Created new tasks in gulpfile to handle modules and their naming conventions, converted gulpfile toward ES6.
- Updated webpack.config with new export specifically for modules and their naming conventions.
- Added new script to package.json specifcally for modules and added that command to build-sass and build-prodution so modules get rebuilt when the other sass does.
- Added an initial version of the new monospace font ( Its ticket: #2700 ).

## Impacted areas of the application
- The box can be seen on the raising and spending bythenumbers pages.
- Updated gulpfile and webpack.config.

## Screenshots
![image](https://user-images.githubusercontent.com/26720877/55813728-4a184700-5abb-11e9-8210-7242e06cb161.png)

## Related PRs
branch | PR
------ | ------ 
feature/2562/embed-bar-charts-election-map-homepage | #2748 
feature/2744-aggregate-totals-box | #2768 

## How to test
- Set the feature flag environment variable `FEC_FEATURE_AGGR_TOTS=1`
- Go to either [raising](http://localhost:8000/data/raising-bythenumbers/) or [spending](http://localhost:8000/data/spending-bythenumbers) bythenumbers.
- The aggregate totals box should only appear after the rest of the page has loaded and only if there's data to display (give it about a second).
- When the animation is complete, check the console log to make sure the query response results total_disbursements (or total_receipts) match the dollar figure on the screen at the end of the animation.
- Change a pull-down, let the animation tick, compare the value with the console values.
- Change the other pull-down, let the animation play, compare the value again.
- Make sure the description text updates for each pull-down change.
- Is the monospace number font applied? (The ones should have more space around them and the dollar sign, commas, and decimal shouldn't jitter)

Notes:
- If it's set to animate the build
  - the number of animation steps/ticks is randomized each time the page is loaded but they should remain the same through each pull-down change.
  - the value increment for the first animation (effectively building from $0) should be ~$100 per animation step/timer tick.
- When switching from one value to the next, the numbers should start ticking into place starting on the right and work their way to the left, one place at a time.
____
